ecssd_agent.go

```diff
@@ -171,49 +169,61 @@ func getDNSHostedZoneId() (string, error) {
 	return "", err
 }
 
-func createDNSRecord(serviceName string, dockerId string, port string) error {
+func createDNSRecord(serviceName string) error {
 	r53 := route53.New(session.New())
+	// This API Call looks for the Route53 DNS record for this service to update
+	paramsList := &route53.ListResourceRecordSetsInput{
+		HostedZoneId:          aws.String(configuration.HostedZoneId), // Required
+		MaxItems:              aws.String("10"),
+		StartRecordName:       aws.String(serviceName + "." + DNSName),
+		StartRecordType:       aws.String(route53.RRTypeA),
+	}
+	resp, err := r53.ListResourceRecordSets(paramsList)
+	logErrorNoFatal(err)
+	if err != nil {
+		return err
+	}
+	// Merge the A records.
+	aValues := resp.ResourceRecordSets[0].ResourceRecords
+	if aValues == nil {
+		aValues = make([]*route53.ResourceRecord, 1)
+	}
+	// the private IPv4 address of the machine providing the service
+	rr := route53.ResourceRecord{Value: aws.String(configuration.LocalIp)}
+	aValues = append(aValues, &rr)
+
 	// This API call creates a new DNS record for this service
 	params := &route53.ChangeResourceRecordSetsInput{
 		ChangeBatch: &route53.ChangeBatch{
 			Changes: []*route53.Change{
 				{
-					Action: aws.String(route53.ChangeActionCreate),
+					Action: aws.String(route53.ChangeActionUpsert),
 					ResourceRecordSet: &route53.ResourceRecordSet{
 						Name: aws.String(serviceName + "." + DNSName),
 						// It creates an A record with the name of the service
 						Type: aws.String(route53.RRTypeA),
-						ResourceRecords: []*route53.ResourceRecord{
-							{
-								// the private IPv4 address of the machine providing the service
-								Value: aws.String(configuration.LocalIp),
-							},
-						},
-						SetIdentifier: aws.String(dockerId),
-						// TTL=0 to avoid DNS caches
-						TTL:           aws.Int64(defaultTTL),
-						Weight:        aws.Int64(defaultWeight),
+						ResourceRecords: aValues,
+						TTL: aws.Int64(defaultTTL),
```

Comment:
The program runs as a daemon and create/delete get called when containers are started or stopped.